cachito/workers/pkg_managers/general_js.py

```diff
@@ -93,7 +94,23 @@ def download_dependencies(request_id, deps):
         for dep_batch in deps_batches:
             log.debug(f"Downloading the following npm dependencies: {', '.join(dep_batch)}")
             npm_pack_args = ["npm", "pack"] + dep_batch
-            run_cmd(npm_pack_args, run_params, "Failed to download the npm dependencies")
+            output = run_cmd(npm_pack_args, run_params, "Failed to download the npm dependencies")
+
+            # Move dependencies to their respective folders
+            # Make tuples out of dependency tarball and dep_identifier
+            # e.g., ('angular-animations-8.2.0.tgz', '@angular/animations@8.2.0')
+            tuples = list(zip(output.split(), dep_batch))
+            for dep_pair in tuples:
+                tarball = dep_pair[0]  # e.g., angular-animations-8.2.0.tgz
+                dir_path = dep_pair[1].rsplit("@", 1)[0]  # e.g., @angular/animations
+                # Create the target directory for dependency
+                dep_dir = bundle_dir.npm_deps_dir.joinpath(*dir_path.split("/", 1))
+                dep_dir.mkdir(exist_ok=True, parents=True)
+                # Move dependency into the target directory
+                shutil.move(
+                    os.path.join(str(bundle_dir.npm_deps_dir), tarball),
+                    os.path.join(str(dep_dir), tarball),
```

Comment:
Could you check to see if the following works?
```suggestion
                    bundle_dir.npm_deps_dir.joinpath(tarball),
                    dep_dir.joinpath(tarball),
```